checkST2110.js

```diff
@@ -325,20 +403,69 @@ const test_20_71_1 = (sdp, params) => {
   let streams = sdp.split(/[\r\n]m=/);
   let errors = [];
   if (params.videoOnly) {
-    for ( let s = 1 ; s < streams.length ; s++ ) {
+    for (let s = 1; s < streams.length; s++) {
       if (!streams[s].startsWith('video')) {
         errors.push(new Error(`Stream ${s}: Media type is not 'video' and video only files are in test, as per SMPTE 2110-20 Section 7.1.`));
       }
     }
+  } else {
+    console.log("Test Skipped: Test ST 2110-20 Section 7.1 Test 1 - Use --videoOnly to check for all video.");
   }
+
+  if (params.videoOnly && params.verbose && errors.length == 0)
+    console.log("Test Passed: Test ST 2110-20 Section 7.1 Test 1 - All streams are video");
+
   return errors;
 };
 
+// Test ST 2110-10 Section 6,2 Test 1 - For all video streams, check video params
+const test_10_62_1 = (sdp, params) => {
+  let errors = [];
+  let lines = splitLines(sdp);
+  videoPresent = false;
+
+  for (let x = 0; x < lines.length; x++) {
+    if (!lines[x].startsWith('m=video')) {
+      continue;
+    }
+    videoPresent = true;
+    let videoMatch = lines[x].match(videoPattern);
+    if (!videoMatch) {
+      errors.push(new Error(`Line ${x + 1}: Found a media description for video with a pattern that is not acceptable.`));
+      continue;
+    }
+    // Check port number - SMPTE 2110-10 Section 6.2 says shall be UDP, so assume 0-65535
+    let port = +videoMatch[1];
+    if (isNaN(port) || port < 0 || port > 65535) {
+      errors.push(new Error(`Line ${x + 1}: RTP video stream description with invalid port '${port}', with reference to ST 2110-10 Section 6.2 'shall use UDP'.`));
+    }
+    // Check RTP type - SMPTE 2110-10 Section 6.2 says shall be RTP, no allowance for SRTP
+    if (videoMatch[3] === 'RTP/SAVP') {
+      errors.push(new Error(`Line ${x + 1}: SRTP protocol is not allowed by SMPTE ST 2110-10 Section 6.2.`));
+    }
+    // Check dynamic range - assume 2110-20 is always dynamic
```

Comment:
I can't quite tell if you mean this test case to check ST 2110-10 Section 6.2, or something from ST 2110-20?
If this is checking ST 2110-10 Section 6.2, then it says "dynamic [...] unless a fixed payload type designation exists".

Reply:
The Original code called out a 2110-20 Section 7.1 test that checked the m=video line for udp port, rtp and dynamic  payload type.  This test is common for 2110-20 and 2110-22 but is not described in 2110-20 section 7.1.  Closest I could see that was specified was in section 6.2 of 2110-10 so I renamed it to avoid confusion and avoid duplicating it for 2110-22.

I also added the check to verify there actually is a m=video line in the SDP

Original code:

```
// Test ST 2110-20 Section 7.1 Test 2 - For all video streams, check video params
const test_20_71_2 = sdp => {
  let errors = [];
  let lines = splitLines(sdp);
  for ( let x = 0 ; x < lines.length ; x++ ) {
    if (!lines[x].startsWith('m=video')) {
      continue;
    }
    let videoMatch = lines[x].match(videoPattern);
    if (!videoMatch) {
      errors.push(new Error(`Line ${x + 1}: Found a media description for video with a pattern that is not acceptable.`));
      continue;
    }
    // Check port number - SMPTE 2110-10 Section 6.2 says shall be UDP, so assume 0-65535
    let port = +videoMatch[1];
    if (isNaN(port) || port < 0 || port > 65535) {
      errors.push(new Error(`Line ${x + 1}: RTP video stream description with invalid port '${port}', with reference to ST 2110-10 Section 6.2 'shall use UDP'.`));
    }
    // Check RTP type - SMPTE 2110-10 Section 6.2 says shall be RTP, no allowance for SRTP
    if (videoMatch[3] === 'RTP/SAVP') {
      errors.push(new Error(`Line ${x + 1}: SRTP protocol is not allowed by SMPTE ST 2110-10 Section 6.2.`));
    }
    // Check dynamic range - assume 2110-20 is always dynamic
    let payloadType = +videoMatch[4];
    if (isNaN(payloadType) || payloadType < 96 || payloadType > 127) {
      errors.push(new Error(`Line ${x + 1}: Dynamic payload type expected for SMPTE 2110-defined video.`));
    }
  }
  return errors;
};
```

Reply:
> I renamed it to avoid confusion

Ah, part of my confusion was that you don't seem to have just renamed it, you have copied it and renamed it so it looked like a whole new test. Do we need to keep the original version as well?

Reply:
Geeze what a mess that was. I think what happened is I had a TODO to make them common then made the logic common but missed the cleanup to remove the duplications. I followed your suggestion and made those changes and committed.